### PR TITLE
[docs] Add ability to include asset preview image next to code block

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adapts the `RenderedDAG` component from the Dagster website to display alongside a code block in the docs.

Can be clicked to expand to the full width of the docs page. Hides automatically on mobile.

Right now, to add a DAG next to a code block, the image path must be included as a prop of the code block itself. A stacked diff w ill create a tool which will assist with generating these images.

See Vercel deployment: https://dagster-git-benpankow-display-assets-next-to-code-elementl.vercel.app/getting-started

```
`'`python dagimage=/images/asset-screenshots/my_assets.svg
import csv
import requests
from dagster import asset

...

`'`
```

## Future Work

It could be nice to by default zoom to a specific asset or set of assets, or highlight a specific asset.

## Demo

https://user-images.githubusercontent.com/10215173/190512004-d38bfe8d-43c9-4cb1-90b8-2d244ec5b0e5.mov

## Test Plan

Vercel, tested locally

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9714** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9714" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9714?utm_source=stack-comment).

File: docs/content/getting-started.mdx
Comment (benpankow) [inline docs/content/getting-started.mdx:7]: reminder to remove before merge

File: docs/next/components/mdx/MDXComponents.tsx
Comment (hellendag) [inline docs/next/components/mdx/MDXComponents.tsx:345]: It's pretty hacky to reach into the props of a child component -- could you instead provide `dagImage` as a separate prop, and leave `children` as is?

File: docs/next/components/mdx/MDXComponents.tsx
Comment (hellendag) [inline docs/next/components/mdx/MDXComponents.tsx:342]: It would be nice to have real props defined for this component, even if it's just `React.HTMLProps<HTMLPreElement>`.

File: docs/next/components/mdx/MDXComponents.tsx
Comment (hellendag) [inline docs/next/components/mdx/MDXComponents.tsx:412]: With a props type defined for the component, the `any` here should be removable.

File: docs/next/components/mdx/RenderedDAG.tsx
Comment (hellendag) [inline docs/next/components/mdx/RenderedDAG.tsx:117]: Is this a brand new component, or copied from somewhere? If it's new, we shouldn't really be using `React.Component` at this point, and all of this should be achievable with a function component.

File: docs/next/components/mdx/RenderedDAG.tsx
Comment (hellendag) [inline docs/next/components/mdx/RenderedDAG.tsx:371]: Is styled-components used in the docs site? I don't see it in package.json.

File: docs/next/components/mdx/RenderedDAG.tsx
Comment (hellendag) [inline docs/next/components/mdx/RenderedDAG.tsx:43]: `// 20`?

**Review by hellendag — CHANGES_REQUESTED** [review]
Requesting changes for the child access pattern, a couple questions inline otherwise.

**Review by erinkcochran87 — COMMENTED** [review]
(I was testing this locally and pushed up a pull from master - sorry about that! I was trying to resolve an error.)

This looks great - I think it'll be really helpful for solidifying understanding of this concept. A few thoughts:

- It looks like this component doesn't pull in external snippets from `examples`, the way `python file=[file]` would - is that correct? If so, that means the code isn't under test the way other snippets are. I'll defer to @yuhan on this, but IIRC we're trying to keep all the snippets under test
- The code block currently has a horizontal scroll (as it should), but no vertical. Do we want to add a max height and enable vertical scrolling for this? I'm thinking of how this might look with longer code snippets
- I think adding some sort of default tooltip/caption/text to this component that explains the available actions (zoom, double-click to center, etc) would be helpful

I really love the side-by-side presentation of this and think there's more we can do with this approach to formatting in the future. For example, having copy and code side-by-side - especially in tutorials and guides - could be really helpful.

**Comment by benpankow** [comment]
> (I was testing this locally and pushed up a pull from master - sorry about that! I was trying to resolve an error.)
> 
> This looks great - I think it'll be really helpful for solidifying understanding of this concept. A few thoughts:
> 
> * It looks like this component doesn't pull in external snippets from `examples`, the way `python file=[file]` would - is that correct? If so, that means the code isn't under test the way other snippets are. I'll defer to @yuhan on this, but IIRC we're trying to keep all the snippets under test

The subsequent PR #9764 adds a command which reads from a snippet file and generates an asset graph image from it. This way, we can update the images whenever we update the snippets (and the snippets themselves will stay under test).

> * The code block currently has a horizontal scroll (as it should), but no vertical. Do we want to add a max height and enable vertical scrolling for this? I'm thinking of how this might look with longer code snippets

Right now the component will just expand vertically the longer the code snippet is (the preview pane will grow correspondingly).

> * I think adding some sort of default tooltip/caption/text to this component that explains the available actions (zoom, double-click to center, etc) would be helpful

I think that could be good, will defer if others have a preference on how this should look.

File: docs/next/components/mdx/MDXComponents.tsx
Comment (benpankow) [reply docs/next/components/mdx/MDXComponents.tsx:345]: Yeah, I'd like to resolve this. I'm not confident in the process in which MDX is being converted into React components - somehow the code block is being transformed into a `pre` tag, which is then being wrapped into our `Pre` component. Ideally instead we would just convert the `pre` tag or be able to move some of its props to our `Pre` component.

File: docs/next/components/mdx/RenderedDAG.tsx
Comment (benpankow) [reply docs/next/components/mdx/RenderedDAG.tsx:117]: Ah, yeah, it's copied from https://github.com/dagster-io/dagster-website/blob/master/components/RenderedDAG.tsx, as with most of this component.

File: docs/next/components/mdx/RenderedDAG.tsx
Comment (benpankow) [reply docs/next/components/mdx/RenderedDAG.tsx:371]: ah no, this is a casualty of copypasta

File: docs/next/components/mdx/MDXComponents.tsx
Comment (benpankow) [reply docs/next/components/mdx/MDXComponents.tsx]: Alright... I found /a/ solution: https://github.com/dagster-io/dagster/pull/9714/commits/860d0228c72c46222fd13866e5f7023bd98d699d

It's a bit gross, but I think it's the nicest solution we'll get.

Right now, a code block in MDX (with three backticks) is converted into a <code> tag wrapped in a <pre> tag, which are nicely styled. However, the <code> tag contains all of the props (e.g. the dag image).

The change in the above commit takes the generated <pre> tag and no-ops the styling. Then, it creates a new <pre> tag in our <code> tag replacement, once we know whether the dag image is set or not.

**Review by yuhan — COMMENTED** [review]
how do you feel about updating the docs lint to reduce the line length. on the mac screen, the code snippets are still being cut, i.e. people need to scroll to see half of the code. but we probably don't want to do that for the code snippets without dagimage, which is tricky for linter to tell.

alternatively, we could make the font even smaller -- the ones on the io site have much smaller fonts. we could specify font size based on "if dagimage"

File: docs/content/getting-started.mdx
Comment (bengotow) [inline docs/content/getting-started.mdx:7]: This is a really slick syntax 🙌

File: docs/next/components/mdx/RenderedDAG.tsx
Comment (bengotow) [inline docs/next/components/mdx/RenderedDAG.tsx:58]: I wonder if we could go ahead and remove the `isAssetGraph` prop? I could see wiring it up to whether or not the image URL contains `-assets` or something like that, but it looks like it's a no-op now that this is commented out (which makes sense, all the buttons could be confusing in the docs)

**Review by bengotow — APPROVED** [review]
This is super cool. I'm glad this could be adapted to the docs!

**Comment by benpankow** [comment]
> how do you feel about updating the docs lint to reduce the line length

will do in follow-up!

**Pull request merged**
